### PR TITLE
Sprint2 (#47)

### PR DESCRIPTION
**Pull request by abasatwar**
### Description
Visualization implementatation for multiple dimension , metrics according to #787 
is implemented for gauge , bar chart and enhancements of heatmap, historgram,teeamap, ui, etc is included in this PR.

**Issues Resolved**
[Enhancement]: Bar chart - Visualizations config panel features v1 #792
[ENHANCEMENT] gauge chart #860
[ENHANCEMENT2] time series #832
Treemap colorscale #820
[FEATURE]: [Visualizations] Sort sectors #789
[ENHANCEMENT]: Updating Visualizations Preview functionality #826
[BUG]: Restrict user to select duplicate fields in dimension and metrics section #827
[ENHANCEMENT-2]: Heatmap visualization config panel features v1 #847
[ENHANCEMENT]: Histogram visualization config panel features v1 #795
[BUG]: Input fields value lag while typing after removing preview functionality #861
[Enhancement]: Bar chart - Visualizations config panel features v1 #792
[FEATURE]:[Visualizations]"No Data Found" screen size on Visualizations page #810

**Check List**
 - [x] New functionality includes testing.
- [x] Commits are signed per the DCO using --signoff
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx [hunk 1]
```diff
@@ -44,56 +44,78 @@ export const ConfigChartOptions = ({
...
       currentSchemas &&
       currentSchemas.map((schema, index) => {
-        let params = {};
+        let params = {
+          title: schema.name,
+          vizState,
+          ...schema.props,
+        };
         const DimensionComponent = schema.component || PanelItem;
         if (schema.eleType === 'palettePicker') {
```
Comment (mengweieric) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx:53]: Could we consider to optimize code here for this set of long if/else branches, maybe with switch?

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx [hunk 2]
```diff
@@ -33,54 +33,75 @@ export const ConfigLineChartStyles = ({
 
   /* To update the schema options based on current style mode selection */
   const currentSchemas = useMemo(() => {
-    if (!vizState?.style || vizState?.style === "lines") {
```
Comment (mengweieric) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx:36]: Similar here, please scan through all these similar files with some optimizations

File: dashboards-observability/public/components/visualizations/charts/bar/bar.tsx [hunk 3]
```diff
@@ -10,80 +10,166 @@ import { LONG_CHART_COLOR, PLOTLY_COLOR } from '../../../../../common/constants/
 import { AvailabilityUnitType } from '../../../event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_availability';
```
Comment (mengweieric) [inline dashboards-observability/public/components/visualizations/charts/bar/bar.tsx:10]: Could we adjust the spaces in this file? I saw some statements are side to side, and some of them are having to empty lines. Please scan through to reformat.

**Comment by mengweieric** [comment]
Please also add comments to some of the logics like dimension/metrics decisions/data transformation etc. for each viz  thanks

**Review by anirudha — APPROVED** [review]

File: dashboards-observability/.cypress/integration/1_event_analytics.spec.js [hunk 4]
```diff
@@ -21,10 +21,29 @@ import {
   renderTreeMapchart,
   renderPieChart,
   renderLineChartForDataConfig,
-  DataConfigLineChart
+  renderDataConfig,
```
[the remark this answers is not in the document]
Comment (pratibhapandey16) [reply dashboards-observability/.cypress/integration/1_event_analytics.spec.js:24]: Yes Eric we have already divided the event file into smaller ones on the basis of different charts and made a new folder for Visualizations charts. The code is still to be merged.
Please refer to the image below and suggest if any change is required.
image

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx (hunk 1 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx:53]: Sure, Thanks @mengweieric, I will do these changes in a separate PR for sprint2.
Created an issue for the same here: https://github.com/opensearch-project/observability/issues/906

File: dashboards-observability/common/constants/explorer.ts [hunk 5]
```diff
@@ -130,3 +132,22 @@ export const ROW_DENSITIES = [
 ];
 
 export const HEADER_HEIGHT = 35;
+
+// gauge chart default parameters
+export interface DefaultGaugeChartParametersProps {
```
[the remark this answers is not in the document]
Comment (rinku-kumar-psl) [reply dashboards-observability/common/constants/explorer.ts:137]: Sure @mengweieric , Thanks!

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_panel_option_gauge.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import { DefaultGaugeChartParameters } from '../../../../../../../../common/constants/explorer';
+
+const helpText = `Limit number of gauges.`;
+
+export const ConfigPanelOptionGauge = ({
+  visualizations,
+  vizState,
+  panelOptionsValues,
+  handleConfigChange,
+}: any) => {
```
[the remark this answers is not in the document]
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_panel_option_gauge.tsx:17]: Sure @mengweieric, We are started adding types and will create a separate PR when it's completed.

File: dashboards-observability/public/components/visualizations/charts/bar/bar.tsx (hunk 3 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/visualizations/charts/bar/bar.tsx:10]: Thanks @mengweieric, We will scan through the files and do the required changes.

**Comment by abasatwar** [comment]
hi @mengweieric  ,
As you requested to earlier , here is list of PR(Internal) where above code is reviewed individually.
https://github.com/abasatwar/observability/pull/61
https://github.com/abasatwar/observability/pull/58
https://github.com/abasatwar/observability/pull/55	
https://github.com/abasatwar/observability/pull/56
https://github.com/abasatwar/observability/pull/54
And code is combined to avoid multiple conflicts.
CC: @spattnaik , @anirudha 

Thanks,
Amruta

**Pull request merged**